src/hotspot/share/gc/shenandoah/shenandoahConcurrentMark.cpp

```diff
@@ -119,8 +119,7 @@ class ShenandoahFinalMarkingTask : public WorkerTask {
       assert(!heap->has_forwarded_objects(), "Not expected");
 
       ShenandoahMarkRefsClosure<GENERATION> mark_cl(q, rp);
-      ShenandoahSATBAndRemarkThreadsClosure tc(satb_mq_set,
-                                               ShenandoahIUBarrier ? &mark_cl : nullptr);
+      ShenandoahSATBAndRemarkThreadsClosure tc(satb_mq_set, nullptr);
```

Comment:
Looks like `ShenandoahSATBAndRemarkThreadsClosure` can be considerably simplified, now that we do not pass any closure to it.